src/common.speech/DialogServiceAdapter.ts

```diff
@@ -719,6 +721,10 @@ export class DialogServiceAdapter extends ServiceRecognizerBase {
             }
         }
 
+        if (this.terminateMessageLoop) {
```

Comment:
this is set from two other places not just the newly added code, is this handled correctly for those cases as well?

Reply:
Yes, I added those in an early mitigation attempt for this same issue manifesting in another way.